swap/swap.go

```diff
@@ -59,15 +63,38 @@ func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 	if err != nil && err != state.ErrNotFound {
 		return
 	}
+
+	//check if balance with peer is over the disconnect threshold
+	if s.balances[peer.ID()] >= s.disconnectThreshold {
+		//if so, return error in order to abort the transfer
+		return fmt.Errorf("balance for peer %s went over the disconnect threshold %v", peer.ID().String(), s.disconnectThreshold)
+	}
+
 	//adjust the balance
 	//if amount is negative, it will decrease, otherwise increase
 	s.balances[peer.ID()] += amount
+
 	//save the new balance to the state store
 	peerBalance := s.balances[peer.ID()]
 	err = s.stateStore.Put(peer.ID().String(), &peerBalance)
+	if err != nil {
+		return
+	}
 
 	log.Debug(fmt.Sprintf("balance for peer %s: %s", peer.ID().String(), strconv.FormatInt(peerBalance, 10)))
-	return err
+
+	//check if balance with peer is over the payment threshold
+	if peerBalance >= s.paymentThreshold {
+		//if so, send cheque request message with the current peer balance as its amount
```

Comment:
So this would only work if the balance is positive, above the threshold. This means it only works when the local node is the creditor and the remote peer got indebted. Which is nice because we decided that it is the debtor who sends the `ChequeRequestMsg`.

But the balance can also cross the threshold in a negative sense. If the local node is the debtor,, then the balance is negative to you peer.  Thus the threshold would be crossed in a negative sense.

In any case, probably in that sense we would take no "real" action, as again, it is the creditor who sends the `ChequeRequestMsg`. But probably we should at least **log** this situation and **warn** the user (or do something more sophisticated later) that he incurred into the payment threshold with a remote peer.

Reply:
I concur. Only the positive case was added due to the cheque-issuing flow we have defined.

I will add a log for the negative case.